### PR TITLE
[Doc]: update doc for pytorch2onnx and onnx2tensorrt

### PR DESCRIPTION
**Pull request by QingChuanWS**
This PR focus on add new content related to batches infer and dynamic shapes in pytorch2onnx.md and onnx2tensorrt.md.

### CODE REVIEW

File: docs/tutorials/onnx2tensorrt.md [hunk 1]
```diff
@@ -69,18 +69,19 @@ python tools/deployment/onnx2tensorrt.py \
...
 
-|    Model    |                        Config                        | Status |
-| :---------: | :--------------------------------------------------: | :----: |
-|     SSD     |             `configs/ssd/ssd300_coco.py`             |   Y    |
-|    FSAF     |        `configs/fsaf/fsaf_r50_fpn_1x_coco.py`        |   Y    |
-|    FCOS     |   `configs/fcos/fcos_r50_caffe_fpn_4x4_1x_coco.py`   |   Y    |
-|   YOLOv3    |  `configs/yolo/yolov3_d53_mstrain-608_273e_coco.py`  |   Y    |
-|  RetinaNet  |   `configs/retinanet/retinanet_r50_fpn_1x_coco.py`   |   Y    |
-| Faster-RCNN | `configs/faster_rcnn/faster_rcnn_r50_fpn_1x_coco.py` |   Y    |
+|    Model    |                        Config                        | Dynamic Shape | batch inference | Note |
```
Comment (RunningLeon) [inline docs/tutorials/onnx2tensorrt.md:72]: Current we have not fully test dynamic shape and muti-batch features on onnx2tensorrt.

File: docs/tutorials/pytorch2onnx.md [hunk 2]
```diff
@@ -78,19 +82,32 @@ python tools/deployment/pytorch2onnx.py \
...
+| Faster-RCNN |   `configs/faster_rcnn/faster_rcnn_r50_fpn_1x_coco.py`   |       Y       |        Y        |      |
 
 Notes:
 
 - *All models above are tested with Pytorch==1.6.0 and onnxruntime==1.5.1*
 
+- The parameters of Non Max Suppression in the above onnx model:
+
+  - `nms_pre`: The number of boxes before NMS when pytorch forward inference. In all the above models, `nms_pre` is `1000`.
+  - `deploy_nums`: The number of boxes before NMS when exporting onnx files in torch. This parameter is `0` in all of the above models.
```
Comment (RunningLeon) [inline docs/tutorials/pytorch2onnx.md:101]: should be `deploy_nms_pre`?

File: docs/tutorials/pytorch2onnx.md [hunk 3]
```diff
@@ -78,19 +82,32 @@ python tools/deployment/pytorch2onnx.py \
...
+  - `max_pre_img`: The number of boxes to be kept after NMS. Among the above models, only the `max_pre_img` parameter of faster_rcnn is `1000`, and the other models are all `200`.
+  - `max_output_boxes_pre`: Maximum number of output  boxes per class of NMS. The `max_output_boxes_pre` is `200` in the SSD model; and the parameter is `100` in other models.
```
Comment (RunningLeon) [inline docs/tutorials/pytorch2onnx.md:103]: should it be `max_output_boxes_per_class`?

File: docs/tutorials/pytorch2onnx.md [hunk 4]
```diff
@@ -78,19 +82,34 @@ python tools/deployment/pytorch2onnx.py \
...
+| Faster-RCNN |   `configs/faster_rcnn/faster_rcnn_r50_fpn_1x_coco.py`   |       Y       |        Y        |      |
 
 Notes:
 
 - *All models above are tested with Pytorch==1.6.0 and onnxruntime==1.5.1*
 
+- The parameters of Non Max Suppression in the above onnx model:
+
+  - `nms_pre`: The number of boxes before NMS when pytorch forward inference. In all the above models, `nms_pre` is `1000`.
+  - `deploy_nums_pre`: The number of boxes before NMS when exporting onnx files in torch. This parameter is `0` in all of the above models.
```
Comment (RunningLeon) [inline docs/tutorials/pytorch2onnx.md:101]: fix typo: deploy_nms_pre

File: docs/tutorials/pytorch2onnx.md [hunk 5]
```diff
@@ -78,19 +82,34 @@ python tools/deployment/pytorch2onnx.py \
...
+  - `max_pre_img`: The number of boxes to be kept after NMS. Among the above models, only the `max_pre_img` parameter of faster_rcnn is `1000`, and the other models are all `200`.
```
Comment (RunningLeon) [inline docs/tutorials/pytorch2onnx.md:102]: typo: max_per_img

File: docs/tutorials/pytorch2onnx.md [hunk 6]
```diff
@@ -78,19 +82,34 @@ python tools/deployment/pytorch2onnx.py \
...
+  - `max_output_boxes_pre_class`: Maximum number of output  boxes per class of NMS. The `max_output_boxes_pre` is `200` in the SSD model and is `100` in other models.
```
Comment (RunningLeon) [inline docs/tutorials/pytorch2onnx.md:103]: typo: max_output_boxes_per_class

File: docs/tutorials/pytorch2onnx.md [hunk 7]
```diff
@@ -78,19 +82,34 @@ python tools/deployment/pytorch2onnx.py \
...
+
+- If the deployed backend platform is TensorRT, please add environment variables before running the file:
+
+  ```bash
+  export ONNX_BACKEND=MMCVTensorRT
+  ```
+
+- If you want to use the `--dynamic-shape` parameter in the TensorRT backend to export onnx, please delete the `--simplify` parameter, and vice versa.
```
Comment (RunningLeon) [inline docs/tutorials/pytorch2onnx.md:111]: The Option is `--dynamic-export` in mmdet.

Comment (RunningLeon) [reply docs/tutorials/pytorch2onnx.md:111]: Add reason why simplify and dynamic shape features cannot be used at same time here.

File: docs/tutorials/pytorch2onnx.md [hunk 8]
```diff
@@ -78,19 +82,39 @@ python tools/deployment/pytorch2onnx.py \
...
+
+In the process of exporting the onnx model, we set some parameters for the NMS op to control the number of output bounding boxes. The following will introduce the parameters settings of the NMS op in the supported models:
+
+- `nms_pre`: The number of boxes before NMS when pytorch forward inference. In all the supported models, `nms_pre` is `1000`.
+
+- `deploy_nms_pre`: The number of boxes before NMS when exporting onnx files in torch. This parameter is `0` in all of the supported models.
+
+- `max_per_img`: The number of boxes to be kept after NMS. Among the supported models, only the `max_pre_img` parameter of faster_rcnn is `1000`, and the other supported models are all `200`.
+
+- `max_output_boxes_per_class`: Maximum number of output  boxes per class of NMS. The `max_output_boxes_pre` is `200` in the SSD model and is `100` in other supported models.
```
Comment (RunningLeon) [inline docs/tutorials/pytorch2onnx.md:116]: second typo： max_output_boxes_per_class

File: docs/tutorials/pytorch2onnx.md [hunk 9]
```diff
@@ -78,19 +82,39 @@ python tools/deployment/pytorch2onnx.py \
...
+
+- If you want to use the `--dynamic-export` parameter in the TensorRT backend to export onnx, please delete the `--simplify` parameter, and vice versa.
+
+## The Parameters of Non-Maximum Suppression in ONNX Export
+
+In the process of exporting the onnx model, we set some parameters for the NMS op to control the number of output bounding boxes. The following will introduce the parameters settings of the NMS op in the supported models:
+
+- `nms_pre`: The number of boxes before NMS when pytorch forward inference. In all the supported models, `nms_pre` is `1000`.
+
+- `deploy_nms_pre`: The number of boxes before NMS when exporting onnx files in torch. This parameter is `0` in all of the supported models.
```
Comment (RunningLeon) [inline docs/tutorials/pytorch2onnx.md:112]: This parameter is `0` in all of the supported models. --> Defaults to 0.

File: docs/tutorials/pytorch2onnx.md [hunk 10]
```diff
@@ -78,19 +82,39 @@ python tools/deployment/pytorch2onnx.py \
...
+  export ONNX_BACKEND=MMCVTensorRT
+  ```
+
+- If you want to use the `--dynamic-export` parameter in the TensorRT backend to export onnx, please delete the `--simplify` parameter, and vice versa.
+
+## The Parameters of Non-Maximum Suppression in ONNX Export
+
+In the process of exporting the onnx model, we set some parameters for the NMS op to control the number of output bounding boxes. The following will introduce the parameters settings of the NMS op in the supported models:
+
+- `nms_pre`: The number of boxes before NMS when pytorch forward inference. In all the supported models, `nms_pre` is `1000`.
```
Comment (RunningLeon) [inline docs/tutorials/pytorch2onnx.md:110]: In all the supported models, `nms_pre` is `1000`. --> Defaults to 1000.

File: docs/tutorials/onnx2tensorrt.md [hunk 11]
```diff
@@ -76,11 +76,12 @@ The table below lists the models that are guaranteed to be convertable to Tensor
 |    FCOS     |   `configs/fcos/fcos_r50_caffe_fpn_4x4_1x_coco.py`   |   Y    |
 |   YOLOv3    |  `configs/yolo/yolov3_d53_mstrain-608_273e_coco.py`  |   Y    |
 |  RetinaNet  |   `configs/retinanet/retinanet_r50_fpn_1x_coco.py`   |   Y    |
-| Faster-RCNN | `configs/faster_rcnn/faster_rcnn_r50_fpn_1x_coco.py` |   Y    |
+| Faster_RCNN | `configs/faster_rcnn/faster_rcnn_r50_fpn_1x_coco.py` |   Y    |
```
Comment (ZwwWayne) [inline docs/tutorials/onnx2tensorrt.md:79]: Faster R-CNN

File: docs/tutorials/onnx2tensorrt.md [hunk 12]
```diff
@@ -76,11 +76,12 @@ The table below lists the models that are guaranteed to be convertable to Tensor
 |    FCOS     |   `configs/fcos/fcos_r50_caffe_fpn_4x4_1x_coco.py`   |   Y    |
 |   YOLOv3    |  `configs/yolo/yolov3_d53_mstrain-608_273e_coco.py`  |   Y    |
 |  RetinaNet  |   `configs/retinanet/retinanet_r50_fpn_1x_coco.py`   |   Y    |
-| Faster-RCNN | `configs/faster_rcnn/faster_rcnn_r50_fpn_1x_coco.py` |   Y    |
+| Faster R-CNN | `configs/faster_rcnn/faster_rcnn_r50_fpn_1x_coco.py` |   Y    |
 
 Notes:
 
 - *All models above are tested with Pytorch==1.6.0 and TensorRT-7.2.1.6.Ubuntu-16.04.x86_64-gnu.cuda-10.2.cudnn8.0*
+- *If the deployed model is Faster_RCNN*, please pass `max_shape = None` for the function [delta2bbox](https://github.com/open-mmlab/mmdetection/blob/master/mmdet/core/bbox/coder/delta_xywh_bbox_coder.py#L134) as a workaround.
```
Comment (RunningLeon) [inline docs/tutorials/onnx2tensorrt.md:84]: Specify this workaround could done when using pytorch2onnx.py to convert models.

File: docs/tutorials/pytorch2onnx.md [hunk 13]
```diff
@@ -72,25 +76,49 @@ python tools/deployment/pytorch2onnx.py \
...
+
+  ```bash
+  export ONNX_BACKEND=MMCVTensorRT
+  ```
+
+- If you want to use the `--dynamic-export` parameter in the TensorRT backend to export onnx, please delete the `--simplify` parameter, and vice versa.
+
+## The Parameters of Non-Maximum Suppression in ONNX Export
+
+In the process of exporting the onnx model, we set some parameters for the NMS op to control the number of output bounding boxes. The following will introduce the parameters settings of the NMS op in the supported models, you can set these parameters through `--cfg-options`.
```
Comment (RunningLeon) [inline docs/tutorials/pytorch2onnx.md:112]: . You can set these parameters ....

File: docs/tutorials/pytorch2onnx.md [hunk 14]
```diff
@@ -72,25 +76,49 @@ python tools/deployment/pytorch2onnx.py \
...
+
+- `nms_pre`: The number of boxes before NMS when pytorch forward inference. The default setting is `1000`.
```
Comment (RunningLeon) [inline docs/tutorials/pytorch2onnx.md:114]: remove ` when pytorch forward inference`

**Review by ZwwWayne — APPROVED** [review]

**Pull request merged**